registry/session.go

```diff
@@ -281,6 +281,9 @@ func (r *Session) GetRepositoryData(remote string) (*RepositoryData, error) {
 	// TODO: Right now we're ignoring checksums in the response body.
 	// In the future, we need to use them to check image validity.
 	if res.StatusCode != 200 {
+		if res.StatusCode == 404 {
```

Comment:
I think this should be like 

```
if res.StatusCode == 404 { ...}
else if res.StatusCode != 200 { ... }
```

which reduces nesting = more readable. Also we should be using `http.StatusNotFound` `http.StatusOK` but this was before you I guess. :)